### DISCUSSION TITLE
Did something change recently about placing the 7-Zip source code in third_party folder?

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#165] [General]
I thought before if I didn't place the 7-Zip source in the third_party folder, bit7z wouldn't build, but now it does.  But the cmake output is saying it is pulling 7-Zip 23.01 and putting it in build/_deps, even if the source is in third_party.  So which is it using?

I definitely don't want to use 23.01 yet.  Can I ensure that I am using 22.01?

**Comment by rikyoz** [comment]
Yeah, it's a recent change in bit7z. The project now doesn't use git submodules anymore for managing dependencies like 7-zip. Instead, it uses [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake) to download and configure them in build/_deps.
I did this since users often didn't read the README or the wiki, and hence didn't put 7-zip in the third-party folder or didn't run the `git submodule`/`git clone ... --recurse` command.
Also, managing the dependencies this way is a bit easier, especially when upgrading them.

In any case, I've added the CMake option `BIT7Z_7ZIP_VERSION`, which can be used to choose a different version of 7-zip to be downloaded, like the 22.01 you need.
Alternatively, you can also use the `BIT7Z_CUSTOM_7ZIP_PATH` option, which allows you to specify a custom path to the 7-zip source code, avoiding downloading it to `build/_deps`.

I'm considering whether to check if the `third_party/7-zip/` folder exists and, if it does, continue using it (i.e., setting up `BIT7Z_CUSTOM_7ZIP_PATH` with that path) to avoid any issues.
